### PR TITLE
add wireguard road warrior setup

### PR DESCRIPTION
**Pull request by mimugmail**

### CODE REVIEW

File: source/manual/how-tos/wireguard-client.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
+============================
+WireGuard Road Warrior Setup
+============================
+
+.. Warning::
+    **IMPORTANT NOTE** :
```
Comment (fabianfrz) [inline source/manual/how-tos/wireguard-client.rst]: Remove it - it is already a warning

File: source/manual/how-tos/wireguard-client.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    WireGuard Plugin is still in development, use at your own risk!
+    
+------------
+Introduction
+------------
+
+WireGuard is a simple and fast modern VPN. It aims to be faster and simpler than IPSec. It intends to be
```
Comment (fabianfrz) [inline source/manual/how-tos/wireguard-client.rst]: simple, fast and modern

The compare somehow gives me the feeling that a benchmark is required. Also there is the question about the kernel. OPNsense uses FreeBSD - has it a kernel implementetion too?

File: source/manual/how-tos/wireguard-client.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+via **VPN->WireGuard**.
+
+--------------------------------
+Step 2a - Setup WireGuard Server
+--------------------------------
+
+The setup of a central VPN server is very simple. Just go to tab **Server** and create a new instance.
+Give it a **Name** and set a desired **Listen Port**. If you have more than one server instance be 
+aware that you can use the **Listen Port** only once. For **Tunnel Address** choose a new virtual 
+network to run communication over it, just like with OpenVPN or GRE (e.g. 192.168.0.1/24).
```
Comment (fabianfrz) [inline source/manual/how-tos/wireguard-client.rst]: does not sound well

File: source/manual/how-tos/wireguard-client.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+**Peers** can not be chosen yet since we have not created them yet. 
+After hitting **Save changes** you can reopen the newly created instance, write down your new public
+key and give it to the other side.
```
Comment (fabianfrz) [inline source/manual/how-tos/wireguard-client.rst]: Maybe mention some secure transport methods

File: source/manual/how-tos/wireguard-client.rst [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+After hitting **Save changes** you can reopen the newly created instance, write down your new public
+key and give it to the other side. 
+
+Now go to tab **Endpoints** and add the fist road warrior, give it a **Name**, insert the **Public
+Key** and the **Tunnel Address** (e.g. 192.168.0.2/24). **Endpoint Address** and  **Endpoint Port**
+can be left empty since they are mostly dynamic, now hit **Save changes**.
+
+Go back to tab **Server**, open the instance and choose the newly created endpoint in **Peers**.
+
+Now we can **Enable** the VPN in tab **General** and go on with the setup.
```
Comment (fabianfrz) [inline source/manual/how-tos/wireguard-client.rst]: "continue" may be better

File: source/manual/how-tos/wireguard-client.rst [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+--------------------------------
+
+The development of WireGuard is very dynamic so this howto won't include any screenshots since 
+features are added rapidly or naming might change. 
+If we have OPNsense also at the client side the configuration is similar to step 3a but you have to
+choose a **Tunnel Address** within the range of the server side and exchange public keys after 
+the creation of a new instance. Then networks which should be routed via WireGuard have to be 
+added to your **Tunnel Address** in the endpoint configuration of your client (e.g. 192.168.0.0/24 
+when this is the LAN of WireGuard server). For pushing all network traffic via VPN you can add 
+0.0.0.0/0. If you do this it's important to also specifiy a DNS server which will be added to your
```
Comment (fabianfrz) [inline source/manual/how-tos/wireguard-client.rst]: V6?

**Comment by mimugmail** [comment]
Thanks for your feedback @fabianfrz 👍 Changed some things.

In my 10G lab is was able to achieve Wireguard performance up to 1.9Gbit without any kernel hacks. IPSec is getting up to 2.4Gbit so it's quite fast so far.

Haven't tested v6 yet since the only dialup supporting v6 is the one offering TV to my family ;)

**Comment by mimugmail** [comment]
Just tested now OpenVPN in the same lab, 1.1Gbit ...

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**